src/chrome/content/rules/Images-Amazon.com.xml

```diff
@@ -0,0 +1,58 @@
+<!--
+	Invalid certificate:
+		ec2.images-amazon.com
+		ec4.images-amazon.com
+		ec5.images-amazon.com
+		ec6.images-amazon.com
+		ec8.images-amazon.com
+		ecx.images-amazon.com
+		g-ecx.images-amazon.com
+	
+	Time out:
+		ec1.images-amazon.com
+		ec3.images-amazon.com
+
+	https://trac.torproject.org/projects/tor/ticket/7539:
+		ecx.images-amazon.com
+
+-->
+<ruleset name="Amazon-Images.com">
+
+	<target host="ec1.images-amazon.com" />
+	<target host="ec2.images-amazon.com" />
+	<target host="ec3.images-amazon.com" />
+	<target host="ec4.images-amazon.com" />
+	<target host="ec5.images-amazon.com" />
+	<target host="ec6.images-amazon.com" />
+	<target host="ec8.images-amazon.com" />
+	<target host="g-ecx.images-amazon.com" />
+	<target host="z-ak.images-amazon.com" />
+
+	<!-- Zeitgeist MP3 player https://trac.torproject.org/projects/tor/ticket/7594 -->
+	<exclusion pattern="^http://g-ecx\.images-amazon\.com/.*(?:[mM][pP]3|zeitgeist|digital/music)" />
+		<test url="http://g-ecx.images-amazon.com/images/G/02/zeitgeist/mp3player/sprites-mp3-preview-player-sm-a._V192561915_.gif" />
+		<test url="http://g-ecx.images-amazon.com/images/G/02/zeitgeist/mp3player/swf/zgMp3Player-1.0._V212273708_.swf" />
+		<test url="http://g-ecx.images-amazon.com/images/G/02/zeitgeist/mp3player/mp3-preview-widget-top-rt-corner._V192561915_.gif" />
+		<test url="http://g-ecx.images-amazon.com/images/G/02/zeitgeist/mp3player/dmusic-chart-sprites-1.0._V192561914_.gif" />
+		<test url="http://g-ecx.images-amazon.com/images/G/02/zeitgeist/mp3player/mp3-preview-widget-top-left-corner._V192561912_.gif" />
+
+	<!--
+		Exclusions for Amazon's Zeitgeist MP3 Player:
+
+			https://trac.torproject.org/projects/tor/ticket/7000
+
+		note that this exclusion can't be scoped to
+		the ssl-images-amazon domain, though it could have
+		copied the crazy pattern from the rule above.
+									-->
+		<exclusion pattern="^http://([\w-]+).images-amazon.com/images/G/01/zeitgeist/mp3player" />
+			<test url="http://ec2.images-amazon.com/images/G/01/zeitgeist/mp3player/swf/zgMp3Player-1.0._V212274098_.swf" />
+			<test url="http://ec1.images-amazon.com/images/G/01/zeitgeist/mp3player/swf/zgMp3Player-1.0._V212274098_.swf" />
+
+	<rule from="^http://g-ecx\.images-amazon\.com/"
+		to="https://d1ge0kk1l5kms0.cloudfront.net/" />
+
+	<rule from="^http://(ec[1-8]|z-ak)\.images-amazon\.com/"
+		to="https://images-na.ssl-images-amazon.com/" />
```

Comment:
I'm not happy with forcing everything to the North America CDN, but I have not yet found a better solution.